include/elf_parser.hpp

```diff
@@ -24,105 +25,246 @@
+ * @brief Structure representing an ELF section with its header and data.
+ */
+typedef struct{
+  GElf_Shdr header;             //!< Section header information
+  std::vector<std::byte> data;  //!< Section data bytes
+}section_s;
+
+
+/**
+ * @class ElfParser
+ * @brief Parser for ELF (Executable and Linkable Format) files.
+ * 
+ * This class provides functionality to parse and extract information from ELF
+ * binary files, supporting both 32-bit and 64-bit architectures. It uses the
+ * libelf library to read ELF headers, sections, program headers, and symbol tables.
+ * All sections, program headers, and symbols are loaded upon construction for 
+ * efficient querying.
+ */
 class ElfParser
 {
   public:
     /**
-     * @brief Constructor of the ElfParser object
-     *
-     * @param p_file_name
+     * @brief Constructs an ElfParser and opens the specified ELF file.
+     * 
+     * Opens the ELF file, validates it, and automatically loads all headers:
+     * - Initializes the libelf library with elf_version(EV_CURRENT)
+     * - Opens the file in read-only mode using open()
+     * - Validates the file is a proper ELF object with elf_kind()
+     * - Loads ELF header via m_load_elf_header()
+     * - Loads all section headers and data via m_load_section_header()
+     * - Loads all program headers via m_load_program_header()
+     * - Loads symbol table via m_load_symbol_table()
+     * 
+     * Prints ELF class information (32-bit or 64-bit) to stdout upon successful initialization.
+     * 
+     * @param p_file_name Path to the ELF file to be parsed.
+     * @throws std::runtime_error Caught internally, prints error and calls exit(EXIT_FAILURE) for:
+     *         - ELF library initialization failures
+     *         - elf_begin() failures
+     *         - Non-ELF object files
+     * @throws std::system_error Caught internally, prints error and calls exit(EXIT_FAILURE) for:
+     *         - File open failures
      */
     ElfParser(std::string_view p_file_name);
 
     /**
-     * @brief Destroy the Elf Parser object
+     * @brief Destroys the ElfParser and releases associated resources.
+     * 
+     * Closes the ELF descriptor using elf_end() and the file descriptor using close().
+     * Prints "ELF file closed." confirmation message to stdout.
      */
     ~ElfParser();
 
     /**
-     * @brief Prints the elf header
+     * @brief Retrieves the ELF file header.
+     * 
+     * Returns the parsed ELF header containing:
+     * - Magic number (ident bytes)
+     * - File type, machine architecture, and version
+     * - Entry point address
+     * - Program header and section header offsets
+     * - Flags and various header sizes
+     * 
+     * @return std::expected<GElf_Ehdr, elf_parser_error> The ELF header on success,
+     *         or UNLOADED_ELF_HEADER error if the header was not loaded during construction.
      */
-    void print_elf_header();
+    std::expected<GElf_Ehdr, elf_parser_error> get_elf_header();
```

Comment:
Not an issue just curious, why std::expected vs an exception that wraps the error code